build_tools/github_actions/github_actions_utils.py

```diff
@@ -0,0 +1,99 @@
+"""Utilities for working with GitHub Actions from Python.
+
+See also https://pypi.org/project/github-action-utils/.
```

Comment:
In hindsight, I should have looked for https://pypi.org/project/github-action-utils/ (source here: https://github.com/saadmk11/github-action-utils/blob/main/github_action_utils.py) before writing this... but oh well.

I don't really want to `pip install` a package just for this, but we could take a source dep / fork that code into third_party.

Some other things in there that would be nice to use / fork / rip off:

* Colored logging for warning/error/etc.
* GitHub log groups
* `overwrite_job_summary` in addition to `append_job_summary`

Some things in this new code which that project does _not_ have:

* Informative logging
* Error checking, which is useful when running locally / outside of github actions